### PR TITLE
chore(core): respect PHP error_reporting level in default logger values

### PR DESCRIPTION
**Pull request by jdalsem**

### CODE REVIEW

File: engine/classes/Elgg/Logger.php [hunk 1]
```diff
@@ -66,19 +66,37 @@ class Logger {
...
+
+	/**
 	 * Constructor
 	 *
 	 * @param PluginHooksService $hooks   Hooks service
 	 * @param Context            $context Context service
 	 * @param Printer            $printer Printer
 	 */
-	public function __construct(PluginHooksService $hooks, Context $context, Printer $printer = null) {
+	public function __construct(PluginHooksService $hooks, Context $context, Printer $printer = null, Config $config) {
```
Comment (jeabakker) [inline engine/classes/Elgg/Logger.php]: `$config` must be before the optional `$printer`

File: engine/classes/Elgg/Logger.php [hunk 2]
```diff
@@ -66,19 +66,37 @@ class Logger {
...
+	 * @var Config
+	 */
+	private $config;
+
+	/**
 	 * Constructor
 	 *
 	 * @param PluginHooksService $hooks   Hooks service
 	 * @param Context            $context Context service
 	 * @param Printer            $printer Printer
```
Comment (jeabakker) [inline engine/classes/Elgg/Logger.php]: missing function doc for `$config`

File: engine/classes/Elgg/Logger.php [hunk 3]
```diff
@@ -66,19 +66,37 @@ class Logger {
...
+		
+		if (($php_error_level & E_ALL) == E_ALL) {
+			$this->level = self::INFO;
+		} elseif (($php_error_level & E_NOTICE) == E_NOTICE) {
+			$this->level = self::NOTICE;
+		} elseif (($php_error_level & E_WARNING) == E_WARNING) {
+			$this->level = self::WARNING;
+		} elseif (($php_error_level & E_ERROR) == E_ERROR) {
+			$this->level = self::ERROR;
+		}
```
Comment (jeabakker) [inline engine/classes/Elgg/Logger.php]: does anyone @Elgg/core  know if this can be done nicer??

**Review by jeabakker — DISMISSED** [review]

File: engine/classes/Elgg/Logger.php (hunk 3 above)
Comment (jdalsem) [reply engine/classes/Elgg/Logger.php]: i think this is beautiful as is :)

**Pull request merged**